### ISSUE TITLE
Unsupported API call has been made

### ISSUE THREAD

**Issue by cblegare** [tonybaloney/sphinxcontrib-confluencebuilder#40]
When uploading docs

### Versions

- Confluence 6.0.6 (Trial license)
- sphinxcontrib-confluencebuilder version `0.7.0.dev0` (installed with `pip install git+https://github.com/tonybaloney/sphinxcontrib-confluencebuilder`)

### Bad HTML

I think the issue is there

```html
<ri:page ri:content-title=''build', 'dist', '* ... '.coverage', 'htmlcov''></ri:page>
```

Where there are single quotes enclosed within singles quotes

In this case, here is the `.conf` content

```text
**class setup.Clean(dist, **kw)**

   Custom clean command to tidy up the project.

   ``default_patterns = ['build', 'dist', '* ... '.coverage', 'htmlcov']``
```

which is generated from the following RestructuredText

```rst
.. autoclass:: setup.Clean

   .. autoattribute:: setup.Clean.default_patterns
```

### Traceback

```text
Traceback (most recent call last):
  File "/home/me/stuff/venv/lib/python3.6/site-packages/sphinx/cmdline.py", line 306, in main
    app.build(opts.force_all, filenames)
  File "/home/me/stuff/venv/lib/python3.6/site-packages/sphinx/application.py", line 339, in build
    self.builder.build_update()
  File "/home/me/stuff/venv/lib/python3.6/site-packages/sphinx/builders/__init__.py", line 331, in build_update
    'out of date' % len(to_build))
  File "/home/me/stuff/venv/lib/python3.6/site-packages/sphinx/builders/__init__.py", line 397, in build
    self.write(docnames, list(updated_docnames), method)
  File "/home/me/stuff/venv/lib/python3.6/site-packages/sphinx/builders/__init__.py", line 434, in write
    self._write_serial(sorted(docnames))
  File "/home/me/stuff/venv/lib/python3.6/site-packages/sphinx/builders/__init__.py", line 443, in _write_serial
    self.write_doc(docname, doctree)
  File "/home/me/stuff/venv/lib/python3.6/site-packages/sphinxcontrib/confluencebuilder/builder.py", line 218, in write_doc
    self.publish_doc(docname, self.writer.output)
  File "/home/me/stuff/venv/lib/python3.6/site-packages/sphinxcontrib/confluencebuilder/builder.py", line 226, in publish_doc
    uploaded_id = self.publisher.storePage(title, output, self.parent_id)
  File "/home/me/stuff/venv/lib/python3.6/site-packages/sphinxcontrib/confluencebuilder/publisher.py", line 310, in storePage
    self.rest_client.put('content', page['id'], updatePage)
  File "/home/me/stuff/venv/lib/python3.6/site-packages/sphinxcontrib/confluencebuilder/rest.py", line 111, in put
    raise ConfluenceBadApiError(err)
sphinxcontrib.confluencebuilder.exceptions.ConfluenceBadApiError: ---
An unsupported Confluence API call has been made.

REQ: PUT
RSP: 400
URL: http://confluence.example.com/rest/api/
API: content
---


sphinxcontrib.confluencebuilder error:
---
An unsupported Confluence API call has been made.

REQ: PUT
RSP: 400
URL: http://confluence.example.com/rest/api/
API: content
---
```

### HTTP Response Content

```json
{
	"statusCode": 400,
	"data": {
		"authorized": false,
		"valid": true,
		"errors": [],
		"successful": false
	},
	"message": "Error parsing xhtml: Unexpected character 'b' (code 98) excepted space, or '>' or \" / > \" \r \n at [row,col {unknown-source}]: [92,65]"
}
```

### HTTP Request Json Body

```json
{  
   "id":"7701249",
   "type":"page",
   "title":"setup module",
   "body":{  
      "storage":{  
         "representation":"storage",
         "value":"<h1>setup module</h1>\n\n<p>The <em>setup.py</em> file is a swiss knife for various tasks.</p>\n\n<p>Start by creating a virtual python environment:</p>\n\n<ac:structured-macro ac:name='code' ac:schema-version='1' ac:macro-id='71774dfb-74bb-416d-a681-98a811f1f5fe'><ac:parameter ac:name='linenumbers'>false</ac:parameter><ac:parameter ac:name='language'></ac:parameter><ac:plain-text-body><![CDATA[\n$ python -m venv .venv\n]]></ac:plain-text-body></ac:structured-macro>\n\n<p>You now can use this isolated clean python environment:</p>\n\n<ac:structured-macro ac:name='code' ac:schema-version='1' ac:macro-id='79129376-6dfc-4d1a-927e-906208158f9a'><ac:parameter ac:name='linenumbers'>false</ac:parameter><ac:parameter ac:name='language'></ac:parameter><ac:plain-text-body><![CDATA[\n$ .venv/bin/python --version\nPython 3.5.2\n]]></ac:plain-text-body></ac:structured-macro>\n\n<p>You may also activate it for the current shell.  POSIX shells would use:</p>\n\n<ac:structured-macro ac:name='code' ac:schema-version='1' ac:macro-id='cbb4c495-5cdc-4eae-b8d8-e00c636b6245'><ac:parameter ac:name='linenumbers'>false</ac:parameter><ac:parameter ac:name='language'></ac:parameter><ac:plain-text-body><![CDATA[\n$ . bin/activate\n]]></ac:plain-text-body></ac:structured-macro>\n\n<h2>running tests</h2>\n\n<p>We use <a href='http://doc.pytest.org/en/latest/'>py.test</a> for running tests because it is amazing.  Run it by invoking the simple <em>test</em> alias of <em>setup.py</em>:</p>\n\n<ac:structured-macro ac:name='code' ac:schema-version='1' ac:macro-id='16184a62-4df9-4432-9c4b-0d910227f133'><ac:parameter ac:name='linenumbers'>false</ac:parameter><ac:parameter ac:name='language'></ac:parameter><ac:plain-text-body><![CDATA[\n$ .venv/bin/python setup.py test\n]]></ac:plain-text-body></ac:structured-macro>\n\n<h2>checking code style</h2>\n\n<p>We use <a href='https://flake8.readthedocs.io/en/latest/'>flake8</a> for enforcing coding standards.  Run it by invoking the simple <em>lint</em> alias of <em>setup.py</em>:</p>\n\n<ac:structured-macro ac:name='code' ac:schema-version='1' ac:macro-id='a5e9d2e4-5365-4767-a382-3d0e3392a8ae'><ac:parameter ac:name='linenumbers'>false</ac:parameter><ac:parameter ac:name='language'></ac:parameter><ac:plain-text-body><![CDATA[\n$ .venv/bin/python setup.py test\n]]></ac:plain-text-body></ac:structured-macro>\n\n<h2>building source distirbutions</h2>\n\n<p>Standard <em>sdist</em> is supported:</p>\n\n<ac:structured-macro ac:name='code' ac:schema-version='1' ac:macro-id='99c4df6d-4cb1-4287-be94-011453fa34f4'><ac:parameter ac:name='linenumbers'>false</ac:parameter><ac:parameter ac:name='language'></ac:parameter><ac:plain-text-body><![CDATA[\n$ .venv/bin/python setup.py sdist\n]]></ac:plain-text-body></ac:structured-macro>\n\n<h2>building binary distributions</h2>\n\n<p>Use the <a href='http://pythonwheels.com/'>wheel distribution standard</a>:</p>\n\n<ac:structured-macro ac:name='code' ac:schema-version='1' ac:macro-id='bc7eade5-017f-4b65-bb94-c82fd0ff4571'><ac:parameter ac:name='linenumbers'>false</ac:parameter><ac:parameter ac:name='language'></ac:parameter><ac:plain-text-body><![CDATA[\n$ .venv/bin/python setup.py bdist_wheel\n]]></ac:plain-text-body></ac:structured-macro>\n\n<h2>building html documentation</h2>\n\n<p>First, make sure to have the technical writer\u00e2\u20ac\u2122s requiments:</p>\n\n<ac:structured-macro ac:name='code' ac:schema-version='1' ac:macro-id='2d730bb4-f0be-407d-a5d5-8ca9c1042b27'><ac:parameter ac:name='linenumbers'>false</ac:parameter><ac:parameter ac:name='language'></ac:parameter><ac:plain-text-body><![CDATA[\n$ .venv/bin/python -m pip install -r requirements/_docs.txt\n]]></ac:plain-text-body></ac:structured-macro>\n\n<p>Then, use <em>setup.py</em> to build the documentation:</p>\n\n<ac:structured-macro ac:name='code' ac:schema-version='1' ac:macro-id='ed46c630-7175-4ee7-8868-2943c1399060'><ac:parameter ac:name='linenumbers'>false</ac:parameter><ac:parameter ac:name='language'></ac:parameter><ac:plain-text-body><![CDATA[\n$ .venv/bin/python setup.py docs\n]]></ac:plain-text-body></ac:structured-macro>\n\n<p>A <a href='https://www.gnu.org/software/make/'>make</a> implementation is not required on any platform, thanks to the <ac:link ac:anchor='setup.Documentation'><ac:link-body>setup.Documentation</ac:link-body></ac:link> class.</p>\n\n<p>*<strong>class setup.Documentation(dist, **kw)</strong>*</p>\n\n<p>   Make the documentation (without the <em>Make</em> program).</p>\n\n   <ac:structured-macro ac:name='info' ac:schema-version='1' ac:macro-id='e7e376f2-da07-4f08-9e09-ca4b537731ba'><ac:rich-text-body><p>This command will not allow any warning from <a href='http://www.sphinx-doc.org/'>Sphinx</a>, treating them as errors.</p></ac:rich-text-body></ac:structured-macro>\n\n<h2>cleaning your workspace</h2>\n\n<p>We also included a custom command which you can invoke through <em>setup.py</em>:</p>\n\n<ac:structured-macro ac:name='code' ac:schema-version='1' ac:macro-id='48f53435-21b0-4d61-a15d-e4f964cdb608'><ac:parameter ac:name='linenumbers'>false</ac:parameter><ac:parameter ac:name='language'></ac:parameter><ac:plain-text-body><![CDATA[\n$ .venv/bin/python setup.py clean\n]]></ac:plain-text-body></ac:structured-macro>\n\n<p>The <ac:link ac:anchor='setup.Clean'><ac:link-body>setup.Clean</ac:link-body></ac:link> command is set to clean the following file patterns:</p>\n\n<p>*<strong>class setup.Clean(dist, **kw)</strong>*</p>\n\n<p>   Custom clean command to tidy up the project.</p>\n\n<p>   ``default_patterns = <ac:link><ri:page ri:content-title=''build', 'dist', '* ... '.coverage', 'htmlcov''></ri:page></ac:link>``</p>"
      }
   },
   "space":{  
      "key":"PROJECTKEY"
   },
   "version":{  
      "number":3
   },
   "ancestors":[  
      {  
         "id":"7701242"
      }
   ]
}
```

**Comment by spyoungtech** [comment]
Seems to related to #39 where I had the same issue with there being some markup in a code block.

**Comment by tonybaloney** [comment]
thanks for filing the issue @cblegare I'll hopefully review next week

**Comment by tonybaloney** [comment]
There's an unescaped string in that list, did you just shorten it? I'll have a go at this on another appl

```
**class setup.Clean(dist, **kw)**

   Custom clean command to tidy up the project.

   ``default_patterns = ['build', 'dist', '* ... '.coverage', 'htmlcov']``
```

**Comment by jdknight** [comment]
This issue should now be resolved in db5b916e57898eb86244b4c33a801f5da5c3b984 (available on the master; see specifically 255043422d25f8e437baf3182eafa6464a1ab288). To take advantage of this change, one can pull from `master`. This change should be made available in an upcoming v0.7 release (#29).

**Comment by cblegare** [comment]
Thanks!
